src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -6,13 +6,15 @@
 
```

Comment:
This is marked as `<IsNETCoreAppRef>false</IsNETCoreAppRef>` in .csproj and there is no nuget package either. I think it means there is no way for people out there to call these APIs even if they opt-in in to preview features. 

Do we need to remove `<IsNETCoreAppRef>false</IsNETCoreAppRef>` ?

Reply:
`<IsNETCoreAppRef>false</IsNETCoreAppRef>` is on purpose. The thinking here is to discourage people from using S.N.Quic, which AFAIU is still possible with direct reference to the implementation lib.

However, I do think we'll get rid of all the attributes in this PR, since it's just redundancy to the app context switch.

cc: @jeffhandley

Reply:
Our current general thinking is to encourage people to try preview features if they are interested. More details is in https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md

What makes S.N.Quic preview different from the generic numeric API preview that we would want to discourage one, and encourage the other?

Reply:
Preview feature is HTTP/3, not quic. So we want to encourage users to try quic indirectly - it is certainly not ready for direct consumption.

Reply:
I understand the concern about compatibility. It is why we have invented RequiresPreviewFeatures and the associated analyzer.

RequiresPreviewFeatures attribute and analyzer is the only hoop we have for other preview features. The APIs for those other preview features can and will change in .NET 7 too.

My question is whether there is something fundamentally different about Quic to deserve more hoops or different hoops.

Reply:
>  So we want to encourage users to try quic indirectly - it is certainly not ready for direct consumption.

I read it as that it is not worth it for anybody to look at and try the APIs in System.Net.Quic and provide feedback, we know that these APIs are not good. Is that correct?

If it is the case, I agree that it should stay private, and System.Net.Quic does not need any RequiresPreviewFeaturesAttribute annotations.

Reply:
> I read it as that it is not worth it for anybody to look at and try the APIs in System.Net.Quic and provide feedback, we know that these APIs are not good. Is that correct?

Yes. We basically de-prioritized the QUIC API in order to focus on delivering HTTP3.

Reply:
My take from this discussion is to get rid of the attributes in S.N.Quic, but keep the one for H/3 version in (even though you still can create the version manually without any warnings).
And the AppContext switch will stay as it is.